ballerina/build.gradle

```diff
@@ -974,354 +1030,215 @@ task testExamples() {
-                file.readLines().each {
-                    str = it.trim()
-                    if (str?.trim()) {
-                        if (str[0] != '#') {
-                            fileOut.add(str)
-                        }
-                    }
+            }
+
+            def lines = "${cmdOutAry[0]}".split('\n')
+            lines = lines + "${cmdOutAry[1]}".split('\n')
+            lines.each {
+                str = it.trim()
+                if (str != "Compiling source" && str != "${balName}.bal" && str != "Running executables" && str != ""
+                        && str != "Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8") {
+                    bbeOutput.add(str)
                 }
-                def lines = "${cmdOutAry[0]}".split('\n')
-                lines = lines + "${cmdOutAry[1]}".split('\n')
-                lines.each {
-                    str = it.trim()
-                    if (str != "Compiling source" && str != "${balName}.bal" && str != "Running executables" && str != "") {
-                        bbeOutput.add(str)
-                    }
+            }
+
+            for (int i = 0; i < bbeOutput.size(); i++) {
+                def index = fileOut.findIndexOf { it == bbeOutput[i] }
+                if (index == -1) {
+                    println "Failed due to output did not match. Line ${i}: ${bbeOutput[i]}"
+                    failFlag = true;
                 }
-                def failFlag = false
-                for (int i = 0; i < bbeOutput.size(); i++) {
-                    def index = fileOut.findIndexOf { it == bbeOutput[i] }
-                    if (index == -1) {
-                        failFlag = true;
-                    }
+            }
+        }
+        return failFlag
+    }
+
+    doLast {
+        buildIgnoreList.each { String elements ->
+            bbeList.remove("$elements")
+        }
+        def failedBuildExamples = []
+        def failOutputVerificationExamples = []
+        bbeList.each { String bbe ->
+            if(runIgnoreList.contains(bbe)) {
+                boolean buildFailFlag = buildBBE(bbe)
+                if (buildFailFlag) {
+                    failedBuildExamples.add(bbe)
                 }
-                if (failFlag) {
-                    reportDetails << "output verification failed: ${balName} \n"
+            } else {
+                boolean outputVerificationFailFlag = outputVerification(bbe)
+                if (outputVerificationFailFlag) {
+                    failOutputVerificationExamples.add(bbe)
                 }
             }
         }
+        
+        boolean buildFailure = false
+        File reportDetails = new File("${baseBuildPath}/result.txt")
+        if (!failedBuildExamples.isEmpty()) {
+            println "\n\tThe following BBEs failed when building\n"
+            failedBuildExamples.each {
+                println "\t ${it}"
+                reportDetails << "Build failed: ${it} \n"
```

Comment:
IMO, we don't need to write the result to a file because we print the failed BBEs'. WDYT?

Reply:
+1 fixed in 78aa2bc